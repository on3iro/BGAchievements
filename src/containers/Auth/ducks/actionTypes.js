export const LOGIN_SUBMIT = 'LOGIN_SUBMIT'
export const LOGIN_REQUEST = 'LOGIN_REQUEST'
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS'
export const LOGIN_ERROR = 'LOGIN_ERROR'
export const LOGOUT = 'LOGOUT'
export const REGISTER_SUBMIT = 'REGISTER_SUBMIT'
export const REGISTER_REQUEST = 'REGISTER_REQUEST'
export const REGISTER_SUCCESS = 'REGISTER_SUCCESS'
export const REGISTER_ERROR = 'REGISTER_ERROR'
export const UPDATE_MAX_ACCOUNTS = 'UPDATE_MAX_ACCOUNTS'
