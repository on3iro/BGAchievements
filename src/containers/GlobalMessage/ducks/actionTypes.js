export const GLOBAL_ERROR = 'GLOBAL_ERROR'
export const GLOBAL_MESSAGE = 'GLOBAL_MESSAGE'
export const CLEAR_MESSAGE = 'CLEAR_MESSAGE'
