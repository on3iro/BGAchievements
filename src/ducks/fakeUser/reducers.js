// @flow

import { loop, Cmd } from 'redux-loop'

import {
  INIT,
  USER_FETCH_FAIL,
  USER_FETCH_SUCCESS
} from './constants'

import type {
  Action,
  State
} from './types'

import {
  fetchUser,
  userFetchFailedAction,
  userFetchSuccessfulAction
} from './actions'

const fakeUserReducer = (state: State = {
  loading: false,
  user: null,
  error: null
}, action: Action): State => {
  switch (action.type) {
    case INIT:
      return loop(
        {...state, loading: true},
        Cmd.run(fetchUser, {
          successActionCreator: userFetchSuccessfulAction,
          failActionCreator: userFetchFailedAction,
          args: ['userA']
        })
      )

    case USER_FETCH_SUCCESS:
      return { ...state, user: action.payload, loading: false }

    case USER_FETCH_FAIL:
      return { ...state, error: action.payload, loading: false }

    default:
      return state
  }
}

export {
  fakeUserReducer
}
