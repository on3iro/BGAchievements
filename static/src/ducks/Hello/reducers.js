import { combineReducers, loop, Cmd } from 'redux-loop'

import * as actions from './actions'
import sideeffects from './sideeffects'

const HelloReducer = (state = [], action) => {
  switch (action.type) {
    case 'SAMPLE_DATA_FETCH_START': {
      return loop(state, Cmd.run(sideeffects.fetchSampleData, {
        successActionCreator: actions.fetchSampleDataSuccess,
        errorActionCreator: actions.fetchSampleDataFailure
      }))
    }

    case 'SAMPLE_DATA_FETCH_SUCCESS': {
      return action.payload
    }

    case 'SAMPLE_DATA_FETCH_FAILURE': {
      return []
    }

    default: {
      return state
    }
  }
}

const LoadingReducer = (state = false, action) => {
  switch (action.type) {
    case 'SAMPLE_DATA_FETCH_START': {
      return true
    }

    case 'SAMPLE_DATA_FETCH_SUCCESS':
    case 'SAMPLE_DATA_FETCH_FAILURE': {
      return false
    }

    default: {
      return state
    }
  }
}

export default combineReducers({
  sampleData: HelloReducer,
  loading: LoadingReducer
})
