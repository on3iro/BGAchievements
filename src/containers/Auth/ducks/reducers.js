import { combineReducers } from 'redux'

import * as types from './actionTypes'

export const user = (state = {
  isLoggedIn: false,
  hasRegistered: false,
  token: '',
  data: {},
  error: {}
}, action) => {
  switch (action.type) {
    case types.LOGIN_SUCCESS: {
      const data = action.payload

      return {
        ...state,
        isLoggedIn: true,
        token: data.token,
        data: data.user,
        error: ''
      }
    }

    case types.REGISTER_SUCCESS: {
      return {
        ...state,
        hasRegistered: true
      }
    }

    case types.REGISTER_ERROR:
    case types.LOGIN_ERROR: {
      const error = action.payload

      return {
        ...state,
        error: error
      }
    }

    case types.LOGOUT: {
      return {
        ...state,
        isLoggedIn: false,
        token: '',
        data: {}
      }
    }

    case types.UPDATE_MAX_ACCOUNTS: {
      return {
        ...state,
        data: {
          ...state.data,
          maxAccounts: action.payload
        }
      }
    }

    default: {
      return state
    }
  }
}

export const AuthRequestState = (state = {
  isLoading: false
}, action) => {
  switch (action.type) {
    case types.LOGIN_SUBMIT: {
      return { ...state, isLoading: true }
    }

    case types.REGISTER_SUBMIT: {
      return { ...state, isLoading: true }
    }

    case types.LOGIN_SUCCESS: {
      return { ...state, isLoading: false }
    }

    case types.REGISTER_SUCCESS: {
      return { ...state, isLoading: false }
    }

    case types.REGISTER_ERROR: {
      return { ...state, isLoading: false }
    }

    case types.LOGIN_ERROR: {
      return { ...state, isLoading: false }
    }

    default:
      return state
  }
}

const authReducer = combineReducers({
  user,
  AuthRequestState
})

export default authReducer
