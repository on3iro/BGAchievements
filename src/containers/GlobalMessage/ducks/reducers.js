import { combineReducers } from 'redux'

import * as types from './actionTypes'

export const GlobalError = (state = '', action) => {
  switch (action.type) {
    case types.GLOBAL_ERROR: {
      return action.payload
    }

    case types.CLEAR_MESSAGE: {
      return ''
    }

    default: {
      return state
    }
  }
}

export const GlobalMessage = (state = '', action) => {
  switch (action.type) {
    case types.GLOBAL_MESSAGE: {
      return action.payload
    }

    case types.CLEAR_MESSAGE: {
      return ''
    }

    default: {
      return state
    }
  }
}

const GlobalMessages = combineReducers({
  GlobalError,
  GlobalMessage
})

export default GlobalMessages
