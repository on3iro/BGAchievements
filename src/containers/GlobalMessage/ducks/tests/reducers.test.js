import * as reducers from '../reducers'
import * as types from '../actionTypes'

describe('GlobalError', () => {
  it('should return initial state', () => {
    expect(reducers.GlobalError(undefined, {})).toMatchSnapshot()
  })

  it('should handle GLOBAL_ERROR', () => {
    expect(reducers.GlobalError(undefined, {
      type: types.GLOBAL_ERROR,
      payload: 'Error'
    })).toMatchSnapshot()
  })

  it('should handle CLEAR_MESSAGE', () => {
    expect(reducers.GlobalError('messag', {
      type: types.CLEAR_MESSAGE
    })).toMatchSnapshot()
  })
})

describe('GlobalMessage', () => {
  it('should return initial state', () => {
    expect(reducers.GlobalMessage(undefined, {})).toMatchSnapshot()
  })

  it('should handle GLOBAL_ERROR', () => {
    expect(reducers.GlobalMessage(undefined, {
      type: types.GLOBAL_MESSAGE,
      payload: 'Message'
    })).toMatchSnapshot()
  })

  it('should handle CLEAR_MESSAGE', () => {
    expect(reducers.GlobalMessage('message', {
      type: types.CLEAR_MESSAGE
    })).toMatchSnapshot()
  })
})
