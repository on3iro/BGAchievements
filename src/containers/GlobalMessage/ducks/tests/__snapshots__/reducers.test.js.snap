exports[`GlobalError should handle CLEAR_MESSAGE 1`] = `""`

exports[`GlobalError should handle GLOBAL_ERROR 1`] = `"Error"`

exports[`GlobalError should return initial state 1`] = `""`

exports[`GlobalMessage should handle CLEAR_MESSAGE 1`] = `""`

exports[`GlobalMessage should handle GLOBAL_ERROR 1`] = `"Message"`

exports[`GlobalMessage should return initial state 1`] = `""`
